src/web/mjs/connectors/CocoManHua.mjs

```diff
@@ -35,22 +25,109 @@ export default class CocoManHua extends ZYMK {
     }
 
     async _getPages(chapter) {
-        const images = await super._getPages(chapter);
-        return images.map(image => {
-            return this.createConnectorURI({
-                url: image,
-                referer: new URL(chapter.id, this.url).href
+        const script = `
+        new Promise((resolve, reject) => {
+            let decrypt = function (key, message) {
+                var s = CryptoJS.enc.Utf8.parse(key),
+                t = CryptoJS.AES.decrypt(message, s, {
+                    mode: CryptoJS.mode.ECB,
+                    padding: CryptoJS.pad.Pkcs7,
+                })
+                return CryptoJS.enc.Utf8.stringify(t).toString()
+            }
+            let totalimgs = 0;
+            try {
+                totalimgs = decrypt('hxPLjUZQgfZQT70x', CryptoJS.enc.Base64.parse(window.mh_info.enc_code1).toString(CryptoJS.enc.Utf8));
+                if (totalimgs == '') {
+                    totalimgs = decrypt('EJJFAD6P2BtTrEN3', CryptoJS.enc.Base64.parse(window.mh_info.enc_code1).toString(CryptoJS.enc.Utf8));
+                }
+                totalimgs = parseInt(totalimgs);
+                if (String(totalimgs) == 'NaN') {
+                    totalimgs = decrypt('EJJFAD6P2BtTrEN3', CryptoJS.enc.Base64.parse(window.mh_info.enc_code1).toString(CryptoJS.enc.Utf8));
+                }
+            }
+            catch (error) {
+                totalimgs = decrypt('EJJFAD6P2BtTrEN3', CryptoJS.enc.Base64.parse(window.mh_info.enc_code1).toString(CryptoJS.enc.Utf8));
+                totalimgs = parseInt(totalimgs);
+            }
+            let picz = [];
+            for( let i = 1 ; i <= totalimgs; i++) {
+                picz.push(__cr.getPicUrl(i));
+            }
+            //get key for picture decryption
+            let realKey = CryptoJS.enc.Utf8.parse(decrypt('EhhTOJE2zA9u7vYT', window.image_info.imgKey));
+            resolve({
+                images : picz, key : realKey
             });
         });
+        `;
+        const uri = new URL(chapter.id, this.url);
+        let request = new Request(uri, this.requestOptions);
+        const data = await Engine.Request.fetchUI(request, script, 30000);
+        return data.images.map(image => this.createConnectorURI( {
+            url : new URL(image, this.url).href, key : data.key})
+        );
     }
 
     async _handleConnectorURI(payload) {
         const request = new Request(payload.url, this.requestOptions);
-        request.headers.set('x-referer', payload.referer);
+        request.headers.set('x-referer', this.url);
+        request.headers.set('x-origin', this.url);
         const response = await fetch(request);
-        let data = await response.blob();
-        data = await this._blobToBuffer(data);
-        this._applyRealMime(data);
-        return data;
+        let buffer = '';
```

Comment:
May set buffer to `undefined`, otherwise a developer may infer it is explicitly intended to be a string.